### DISCUSSION TITLE
[FEATURE REQUEST] Show warning when module/state overrides are in use

### DISCUSSION THREAD

**Discussion by timwsuqld** [saltstack/salt#67550] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
When using a module override (`_modules`, `_states`) we can forget that the override is in place when later updates cause issues. This can cause problems with reporting bugs when an issue is fixed in a newer salt version but you forget you had an override so you don't get the update (See https://github.com/saltstack/salt/issues/56124#issuecomment-620901850). It's also an issue when one system admin implements an override, but other system admins aren't aware.

**Describe the solution you'd like**
A clear warning/info message in highstate output, and or in salt-version output. (salt-version output is required for bug reports, so when collecting that it would be clear you have an override in place both for you reporting the bug, and for those looking at your bug report). @boltronics suggested an INFO message at startup, which would require less work to implement.

**Comment by max-arnold** [comment]
It makes sense to also reopen and fix this one: https://github.com/saltstack/salt/issues/16439
